servers/services/src/main/java/org/projectnessie/services/impl/RefLogApiImpl.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.services.impl;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.projectnessie.api.RefLogApi;
+import org.projectnessie.api.params.RefLogParams;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.error.NessieRefLogNotFoundException;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.Content.Type;
+import org.projectnessie.model.ImmutableRefLogResponse;
+import org.projectnessie.model.ImmutableRefLogResponseEntry;
+import org.projectnessie.model.RefLogResponse;
+import org.projectnessie.services.authz.AccessChecker;
+import org.projectnessie.services.config.ServerConfig;
+import org.projectnessie.versioned.Hash;
+import org.projectnessie.versioned.RefLogEntry;
+import org.projectnessie.versioned.RefLogNotFoundException;
+import org.projectnessie.versioned.VersionStore;
+
+public class RefLogApiImpl extends BaseApiImpl implements RefLogApi {
+
+  private static final int MAX_REF_LOG_ENTRIES = 250;
+
+  public RefLogApiImpl(
+      ServerConfig config,
+      VersionStore<Content, CommitMeta, Type> store,
+      AccessChecker accessChecker,
+      Principal principal) {
+    super(config, store, accessChecker, principal);
+  }
+
+  @Override
+  public RefLogResponse getRefLog(RefLogParams params) throws NessieNotFoundException {
+    int max =
```

Comment:
The code here looks very much like the code to retrieve the commit log.
Can it be unified and the common parts implemented in BaseApiImpl?

Reply:
The building different objects here also we don't need filter and get commit log needs reference. hence cannot combine. Not much common code left. They look similar but for different purpose.